### ISSUE TITLE
Getting 3.0.0 released

### ISSUE THREAD

**Issue by vqvu** [caolan/highland#536]
The 3.0.0 branch has been around for a long time, and I don't think any of us ever meant to drag it out for so long. I believe much of this is due to lack of motivation by the maintainers (especially me) to push things forward and get it released.

As it is, there are people [actually using](https://github.com/caolan/highland/issues/458) the 3.0 beta release, and we are getting [requests](https://github.com/caolan/highland/issues/532#issuecomment-245044369) for new features to be forward ported to 3.0. I think it's important that we actually get the thing properly released.

To that end, I propose that we switch the `master` branch to point at 3.x code and relegate 2.x code to a bugfix branch. The hope is that this will provide some form of motivation for people to contribute to 3.x. It would also be easier to make more radical changes to 3.x (e.g., split up `index.js`) if we don't have to worry about merging in 2.x code all the time. What are peoples' thoughts? @svozza? (picking on you since you seem to be around.)

Also, if there anybody reading this who would like to help us get to a 3.0.0 release, please let me know!

**Comment by svozza** [comment]
Yep, I think this is a great idea. I also know I'm one of the people holding it up because of #412. I kind of got stuck on that one and then distracted with other projects but I'll try to do some focused work on it soon. Will we merge your last PR and then release `2.10` before we make the switch?

**Comment by jeromew** [comment]
+1 on this. @vqvu you have done a great job on handling highland. I have been a bit far from the project because of other duties sorry for that.

releasing 3.0 and forgetting about 2.x would be very good for the project. It is not possible to keep maintaining 2.x after the great job you did on 3.0

**Comment by vqvu** [comment]
Thanks both of you for the input! Here's what I've done:

- Released `2.10.0`.
- Forward ported the changes in `2.10` to `3.0` and released `3.0.0-beta.3`.
- Merged `3.0.0` onto `master`. The `3.0.0` branch is now defunct.
- Created a `2.x` branch that tracks 2.x code starting from the `2.10.0` tag.
- Added a note to the `README.md` about the current state.

**Comment by amsross** [comment]
Is this still a goal?

**Comment by vqvu** [comment]
I haven't had the motivation to work on this in a while, partly because I changed jobs, so I'm no longer using highland at work, and partly because there hasn't really been many bugs files against 2.x lately. 

Theres not too much coding work left (the current beta is in a usable state), but there's quite a bit of documentation to write up. Unless someone volunteers to finish up the work though, I wouldn't expect 3.0 to be out in the near term. Sorry.
